integration_test.go

```diff
@@ -108,6 +108,50 @@ func TestWrapUnwrap(t *testing.T) {
 	}
 }
 
+func TestWrapUnwrapWithKeyVersion(t *testing.T) {
+	assert := assert.New(t)
+
+	c, err := NewIntegrationTestClient(t)
+	assert.NoError(err)
+
+	ctx := context.Background()
+
+	keys, err := c.GetKeys(ctx, 0, 0)
+	assert.NoError(err)
+
+	for _, key := range keys.Keys {
+		t.Logf("%+v\n", key)
+	}
+
+	crk, err := c.CreateKey(ctx, "kptest-crk", nil, false)
+	assert.NoError(err)
+	t.Logf("CRK created successfully: id=%s\n", crk.ID)
+
+	wrapWithKeyVersion, err := c.WrapWithKeyVersion(ctx, crk.ID, nil, nil)
+	assert.NoError(err)
+
+	unwrapped, err := c.Unwrap(ctx, crk.ID, []byte(wrapWithKeyVersion.CipherText), nil)
+	assert.EqualValues(unwrapped, wrapWithKeyVersion.PlainText)
+
+	getKeyVersion, err := c.ListKeyVersions(ctx, crk.ID, nil)
+	assert.NoError(err)
+	assert.EqualValues(getKeyVersion.KeyVersion[0].ID, wrapWithKeyVersion.KeyVersionDetails.ID)
```

Comment:
We should be doing this for wrap call. right?

Reply:
We are confirming the KeyVersion returned from Wrap with the List Keyversion Call for the same key to attest that the functionality is working fine !